app/models/team.rb

```diff
@@ -35,11 +35,28 @@ class Team < ApplicationRecord
   has_many :reports, inverse_of: :team, dependent: :destroy
   has_many :datatables_reports,
            class_name: 'Views::Datatables::DatatablesReport'
+  has_many :view_states, as: :viewable, dependent: :destroy
 
   after_commit do
     Views::Datatables::DatatablesReport.refresh_materialized_view
   end
 
+  def current_view_state(user)
+    state = view_states.where(user: user).take
+    state || view_states.create!(user: user, state: default_view_state)
+  end
+
+  def default_view_state
```

Comment:
I just have a question: let's say that we also implement state save for e.g. individual team settings page (where there's a `DataTables`  listing all the users in a team). In this case, we're also storing a state for a `current_user` - `current_team` combo. Wouldn't it make sense to add additional root-level to this JSONB to designate which "view" for the user-object relation the state belongs to?

Reply:
Yeah, it makes sense